src/inference/inference.jl

```diff
@@ -92,24 +92,47 @@ function inference_process_error(error)
 end
 
 function inference_process_error(error, rethrow)
-    if rethrow
-        Base.rethrow(error)
+    if error isa StackOverflowError
+        @error """
+        Stack overflow error detected during inference. This can happen with large model graphs 
+        due to recursive message updates.
+
+        Possible solution:
+        Try using the `limit_stack_depth` inference option. If this does not resolve the issue,
+        please open a GitHub issue at https://github.com/ReactiveBayes/RxInfer.jl/issues and we'll help investigate.
+
+        For more details:
+        • Stack overflow guide: https://reactivebayes.github.io/RxInfer.jl/stable/manuals/sharpbits/stack-overflow-inference/
+        • See `infer` function docs for options
+        """
     end
-    return error, catch_backtrace()
-end
-
-# We want to show an extra hint in case the error is of type `StackOverflowError`
-function inference_process_error(err::StackOverflowError, rethrow)
     @error """
-    Stack overflow error occurred during the inference procedure. 
-    The inference engine may execute message update rules recursively, hence, the model graph size might be causing this error. 
-    To resolve this issue, try using `limit_stack_depth` inference option for model creation. See `?inference` documentation for more details.
-    The `limit_stack_depth` option does not help against over stack overflow errors that might happening outside of the model creation or message update rules execution.
+    We encountered an error during inference, but don't worry - we're here to help! 🤝
+
+    Here are some helpful resources to get you back on track:
+
+    1. Check our Sharp bits documentation which covers common issues:
+       https://reactivebayes.github.io/RxInfer.jl/stable/manuals/sharpbits/overview/
+
+    2. Browse our existing issues - your question may already be answered:
+       https://github.com/ReactiveBayes/RxInfer.jl/issues
+
+    Still stuck? We'd love to help! You can:
+    - Start a discussion for questions and help:
```

Comment:
Super nice!! ✨

```suggestion
    - Start a discussion for questions and help. Feedback and questions from new users is also welcome! If you are stuck, please reach out and we will solve it together.
```

Maybe some more words to fight the idea that GH discussions is only for:
- experts
- bugs/technical problems